File: client/ts/src/storage.ts
import fs from "node:fs";
import os from "node:os";
import path from "node:path";

export function getClientStorageRoot(): string {
  if (process.platform === "win32") {
    const pd = process.env.ProgramData || "C:\\ProgramData";
    const p = path.join(pd, ".RuntimeRepository");
    fs.mkdirSync(p, { recursive: true });
    return p;
  }
  const p = path.join(os.tmpdir(), ".runtime-repository");
  fs.mkdirSync(p, { recursive: true });
  return p;
}

File: client/go/storage.go
package authclient

import (
	"os"
	"path/filepath"
	"runtime"
)

func DefaultClientStorageRoot() string {
	if runtime.GOOS == "windows" {
		pd := os.Getenv("ProgramData")
		if pd == "" {
			pd = `C:\ProgramData`
		}
		return filepath.Join(pd, ".RuntimeRepository")
	}
	return filepath.Join(os.TempDir(), ".runtime-repository")
}
